conda/environment.yml

```diff
@@ -8,13 +8,13 @@ channels:
   - conda-forge
 dependencies:
   - ipyleaflet
-  - jupyter-book=0.13
+  - jupyter-book
   - jupyter-resource-usage
   - jupyterhub-singleuser
-  - jupyterlab=3
+  - jupyterlab
   - jupytext
   - pip
-  - python=3.10
```

Comment:
unpinned major dependencies here (since it's been a while this takes us to jlab=4, jbook=1, python=3.12)!

Reply:
I think we froze those versions back then cause the build was breaking with the 0.14 book version.
Generally I am a fan of pinning major versions (i.e <2.0 for Jupyter Book for instance). Brings less surprises when coming back to this template and having things break with libraries advancing their development.

Reply:
If there is a problem building in the future we can always bring back a pin. I think no pins are better for solving the full environment dependency tree and ensuring we have the latest functionality, but I don't feel particularly strongly... so happy to pin major versions.

Reply:
How about we just pin the book to be <2 and call it good?